cs-rin-ru-enhanced-mod.user.js

```diff
@@ -923,16 +991,227 @@ async function colorizeFriends() {
+    let showResultsAsPosts = document.getElementById("showAsPosts").checked ? "posts" : "topics"; // Display
+    let searchAuthor = document.getElementById("searchAuthor").value; // Author
+    let forumID = "";
+    let topicID = "0";
+
+    // Check the searchScope and parse URL if required
+    if (searchScope === "thisForum") {
+        let urlParams = new URLSearchParams(window.location.search);
+        forumID = urlParams.get("f");
+        if (forumID) {
+            forumID = "&fid%5B%5D=" + forumID;
+        }
+    }
+
+    if (searchScope === "thisTopic") {
+        let urlParams = new URLSearchParams(window.location.search);
+        topicID = urlParams.get("t");
+    }
+
+    window.location.href = `./search.php?keywords=${encodeURIComponent(searchBar.value).replace(/%20/g, "+")}&terms=${searchTerms}&author=${encodeURIComponent(searchAuthor).replace(/%20/g, "+")}${forumID}&sc=${searchSubforums}&sf=${searchLocation}&sk=${sortResultsBy}&sd=${sortOrderBy}&sr=${showResultsAsPosts}&st=${limitToPrevious}&ch=${returnFirst}&t=${topicID}`;
+}
+
+async function specialSearch() {
+    if (options.special_search) {
+        // Get row to insert searchBar
+        const cell = document.querySelector("#menubar > table:nth-child(3) > tbody > tr > td:nth-child(2)");
+        const container = document.createElement("div");
+        container.style.position = "relative";
+        container.style.display = "inline-block";
+
+        // Different locations based on which page the user is on
+        let searchScopeOptions;
+        if (window.location.href.includes("viewtopic.php")) {
+            searchScopeOptions = `
+                <option value="everywhere">Everywhere</option>
+                <option value="thisForum">This forum</option>
+                <option value="thisTopic">This topic</option>
+            `;
+        } else if (window.location.href.includes("viewforum.php")) {
+            searchScopeOptions = `
+                <option value="everywhere">Everywhere</option>
+                <option value="thisForum">This forum</option>
+            `;
+        } else {
+            searchScopeOptions = `
+                <option value="everywhere">Everywhere</option>
+            `;
+        }
+
+        // Getting config values
+        let specialSearchParametersJSON = JSON.parse(options.special_search_parameter);
+        const searchLocationChecked = specialSearchParametersJSON.searchTopicLocation === "titleonly" || specialSearchParametersJSON.searchTopicLocation === "firstpost" ? "checked" : "";
+        const showAsPostsChecked = specialSearchParametersJSON.showResultsAsPosts ? "checked" : "";
+        const searchTermsSelected = specialSearchParametersJSON.searchTermsSpecificity;
+
+        // Creating search bar and search options
+        container.innerHTML = `
```

Comment:
This search panel could also use some padding between panel border and the controls.